miniapp/miniapp.cpp

```diff
@@ -67,7 +69,12 @@ int main(int argc, char** argv) {
             global_policy::set_sizes(options.dry_run_ranks, cells_per_rank);
         }
 
-        banner();
+        // Use a node description that uses the number of threads used by the
+        // threading back end, and 1 gpu if available.
+        hw::node_info nd;
+        nd.num_cpu_cores = threading::num_threads();
+        nd.num_gpus = hw::num_gpus()? 1: 0;
```

Comment:
A `node_description` is a simple description of resources available on a node. This is currently a  count of each resource, currently cpu cores and gpus.
In this context, it is used to tell the `domain_decomposition` how to partition cell groups. We can only support a single GPU per MPI rank at the moment, so we use `min(1, num_gpus);`.

I have changed the statement to
```
nd.num_gpus = hw::num_gpus()>0? 1: 0;
```
make this a little bit clearer.